api/src/database/helpers/schema/utils/preprocess-bindings.ts

```diff
@@ -1,46 +1,72 @@
 import { isString } from 'lodash-es';
 import type { Sql } from '../types.js';
+import type { Knex } from 'knex';
 
 export type PreprocessBindingsOptions = {
 	format(index: number): string;
 };
 
+/**
+ * Preprocess a SQL query, such that repeated binding values are bound to the same binding index.
+ **/
 export function preprocessBindings(
 	queryParams: (Partial<Sql> & Pick<Sql, 'sql'>) | string,
 	options: PreprocessBindingsOptions,
 ) {
 	const query: Sql = { bindings: [], ...(isString(queryParams) ? { sql: queryParams } : queryParams) };
 
+	// bindingIndices[i] is the index of the first occurrence of query.bindings[i]
 	const bindingIndices: number[] = new Array(query.bindings.length);
 
 	for (let i = 0; i < query.bindings.length; i++) {
 		const binding = query.bindings[i];
 		const prevIndex = query.bindings.findIndex((b, j) => j < i && b === binding);
 
-		if (prevIndex !== -1) {
-			bindingIndices[i] = prevIndex;
-		} else {
+		if (prevIndex === -1) {
+			// The first time the value is encountered, set the index lookup to the current index
 			bindingIndices[i] = i;
+		} else {
+			// The value has been encountered before, set the index lookup to the first occurrence index
+			bindingIndices[i] = prevIndex;
 		}
```

Comment:
This is just switching the order of the if statement, as it is more intuitive to read if the first block is the first time a value is encountered (`prevIndex === -1`)